lib/std/posix.zig

```diff
@@ -4276,6 +4276,32 @@ pub fn connect(sock: socket_t, sock_addr: *const sockaddr, len: socklen_t) Conne
     }
 }
 
+pub const GetSockOptError = error{
+    /// The calling process does not have the appropriate privileges.
+    AccessDenied,
+
+    /// The option is not supported by the protocol.
+    InvalidProtocolOption,
+
+    /// Insufficient resources are available in the system to complete the call.
+    SystemResources,
+} || UnexpectedError;
+
+pub fn getsockopt(fd: socket_t, level: i32, optname: u32, noalias optval: ?*anyopaque, noalias optlen: *socklen_t) GetSockOptError!void {
```

Comment:
What is the rationale for this signature instead of doing something similar to `setsockopt()`?

https://github.com/ziglang/zig/blob/8da774afaf4b371156b22dfb242c237ebc66b421/lib/std/posix.zig#L6788

Reply:
not a good one, let me try changing it